protocol/kafka_confluent/v2/message.go

```diff
@@ -0,0 +1,143 @@
+/*
+ Copyright 2023 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+)
+
+const (
+	prefix         = "ce-"
+	contentTypeKey = "Content-Type"
+)
+
+const (
+	KafkaOffsetKey    = "kafkaoffset"
+	KafkaPartitionKey = "kafkapartition"
+	KafkaTopicKey     = "kafkatopic"
+	KafkaMessageKey   = "kafkamessagekey"
+)
+
+var specs = spec.WithPrefix(prefix)
+
+// Message represents a Kafka message.
+// This message *can* be read several times safely
+type Message struct {
+	internal   *kafka.Message
+	properties map[string][]byte
+	format     format.Format
+	version    spec.Version
+}
+
+// Check if Message implements binding.Message
+var (
+	_ binding.Message               = (*Message)(nil)
+	_ binding.MessageMetadataReader = (*Message)(nil)
+)
+
+func NewMessage(msg *kafka.Message) *Message {
+	var contentType, contentVersion string
```

Comment:
nit: want to add `nil` check for `msg` and `msg.TopicPartition.Topic` and `panic()` immediately if someone uses the function wrong?

Reply:
Done.

Reply:
nit: just noticed code comment missing for exported function